ansible_runner/interface.py

```diff
@@ -131,6 +131,10 @@ def run(**kwargs):
     :param resource_profiling_memory_poll_interval: Interval (in seconds) between memory polling for determining memory usage (default: 0.25)
     :param resource_profiling_pid_poll_interval: Interval (in seconds) between polling PID count for determining number of processes used (default: 0.25)
     :param resource_profiling_results_dir: Directory where profiling data files should be saved (defaults to profiling_data folder inside private data dir)
+    :param containerized: Indicates ansible task should be executed using an execution environment (default: false)
```

Comment:
I wonder if "containerization" is actually the right thing to use here. This would seem to be a modification of the process isolation system and doing it this way doesn't make bwrap and container usage mutually exclusive... which it probably should be.

Reply:
@matburt That brings up a question that I've had. How much longer will we need to support both container and non-container based execution? Is that something that I could go ahead and drop in this PR?

Reply:
If we're not going to drop non-container execution, then:

* do we want to make 'containerized' a `process_isolation` / `process_isolation_executable` setting? While containerization is a process isolation mechanism, this seems a little unintuitive.
* or, do we just want to add some logic to disable bwrap when `containerized: True`? I'm guessing this is the middle-ground that we'll end up going with.

Reply:
I am definitely not a fan of changing this option to `containerization`. `ansible-runner run --containerized` makes way more sense than `ansible-runner run --containerization`. Am I missing something?

And I think we should just throw an error if you try to use `--process-isolation` at the same time as `--containerized`.

Reply:
@shanemcd and I caught up about this. We ended up with these options, which folds containerization into the existing process isolation options, and adds a few options specific to containerization:

```
--process-isolation => {true, false}
--process-isolation-executable => {bwrap, podman, docker}
--container-image => defaults to `ansible/ansible-runner`
--container-volume-mounts => list of host:container dir mappings
--container-options => list of any other options you may want to pass to the execution engine
```